### PR TITLE
Update pulumi deps to v3.7.0

### PR DESCRIPTION
**Pull request by lblackstone**
### Related issues (optional)

Fix https://github.com/pulumi/pulumi-kubernetes/issues/1645

### CODE REVIEW

File: sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ public InputList<System.Text.Json.JsonElement> Enum
         public Input<string>? Id { get; set; }
 
         [Input("items")]
-        public InputUnion<Pulumi.Kubernetes.Types.Inputs.ApiExtensions.V1.JSONSchemaPropsArgs, ImmutableArray<System.Text.Json.JsonElement>>? Items { get; set; }
+        public InputUnion<Pulumi.Kubernetes.Types.Inputs.ApiExtensions.V1.JSONSchemaPropsArgs, ImmutableArray<InputJson>>? Items { get; set; }
```
Comment (mikhailshilkov) [inline sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs:103]: It's just two properties and I'm not sure how commonly used they are but I think this is technically breaking...  cc @pgavlin

File: sdk/python/pulumi_kubernetes/networking/v1/_inputs.py [hunk 2]
```diff
@@ -1078,6 +1006,78 @@ def policy_types(self, value: Optional[pulumi.Input[Sequence[pulumi.Input[str]]]
         pulumi.set(self, "policy_types", value)
 
 
+@pulumi.input_type
+class NetworkPolicyArgs:
```
Comment (mikhailshilkov) [inline sdk/python/pulumi_kubernetes/networking/v1/_inputs.py:1010]: It looks like things are just moved around in Python but it's really hard to tell if there are any significant changes here. Do we trust there are none?

Comment (lblackstone) [reply sdk/python/pulumi_kubernetes/networking/v1/_inputs.py:1010]: Yes, I checked, and it was just reordering some of the classes.

File: sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs (hunk 1 above)
Comment (pgavlin) [reply sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs:103]: I _think_ that the new version is "more correct": it _should_ accept a wider set of values than the original definition, and I don't _think_ that user code that passes inputs to these properties will need to change. However, I'm not entirely certain there, and it is technically breaking from the POV of copying one of these fields to a variable that has an explicit type name. Do we have tests that exercise this field?

Comment (mikhailshilkov) [reply sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs:103]: Usually, an `Array<T>` won't convert to `Array<U>` implicitly even if `T` is implicitly convertable to `U`. So

```cs
// compiles
InputUnion<string, ImmutableArray<JsonElement>> works = new JsonElement[] { new JsonElement() }.ToImmutableArray();
// wont compile
InputUnion<string, ImmutableArray<InputJson>> fails = new JsonElement[] { new JsonElement() }.ToImmutableArray();
```

Comment (pgavlin) [reply sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs:103]: Ugh, right

Comment (mikhailshilkov) [reply sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs:103]: Filed https://github.com/pulumi/pulumi/issues/7569

Comment (lblackstone) [reply sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs]: @mikhailshilkov I pulled in @pgavlin's fix, so can you double-check that with the latest commit.

Comment (mikhailshilkov) [reply sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs]: The type has changed from `ImmutableArray` to `InputList` but I _think_ it's okay.

Comment (pgavlin) [reply sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs]: Ah, damn--let me see if I can avoid that change. I'd rather it stick as `ImmutableArray` to minimize the diff.

Comment (pgavlin) [reply sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs]: Updated my PR to fix that type diff.

Comment (lblackstone) [reply sdk/dotnet/ApiExtensions/V1/Inputs/JSONSchemaPropsArgs.cs]: Pulled in @pgavlin's fix, so this should be resolved now.

**Comment by lblackstone** [comment]
@viveklak This is RFR

**Review by viveklak — APPROVED** [review]

**Pull request merged**